src/geo/map/legends/legend-model-base.js

```diff
@@ -54,11 +54,16 @@ var LegendModelBase = Backbone.Model.extend({
 
   reset: function () {
     var defaults = _.omit(this.defaults(),
-      NON_RESETEABLE_DEFAULT_ATTRS
+      NON_RESETEABLE_DEFAULT_ATTRS,
+      this._avoidResetTheseAttributes()
```

Comment:
There's already a mechanism to specify non resettable default attrs (`NON_RESETEABLE_DEFAULT_ATTRS`)... Instead of adding a new mechanism, could we unify them into one? 

```
// in legend-model-base.js
getNonResettableAttrs: function () {
   return ['state', 'visible'];
}
``` 

```
// in a subclass (eg: category-legend-model.js)

getNonResettableAttrs: function () {
  return _.union(
    LegendModelBase.prototype.getNonResettableAttrs.apply(this),
    ['categories']
  );
}

```

Anyway, not resetting these attributes (categories, etc) means that legend models will still be "available" after they are reset... is this what we want?